internal/client/slas.go

```diff
@@ -0,0 +1,60 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/uuid"
+
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.SLAsClient(&SLAsClient{})
+
+// SLAsClient is a client for working with SLAs.
+type SLAsClient struct {
+	hc          *http.Client
+	routePrefix string
+	apiKey      string
+}
+
+// SLAs returns a SLAsClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) SLAs(accountID uuid.UUID, workspaceID uuid.UUID) (api.SLAsClient, error) {
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	if err := validateCloudEndpoint(c.endpoint, accountID, workspaceID); err != nil {
+		return nil, err
+	}
+
+	return &SLAsClient{
+		hc:          c.hc,
+		apiKey:      c.apiKey,
+		routePrefix: getWorkspaceScopedURL(c.endpoint, accountID, workspaceID, "slas"),
+	}, nil
+}
+
+// ApplyResourceSLAs applies SLAs to a resource.
+func (c *SLAsClient) ApplyResourceSLAs(ctx context.Context, resourceID uuid.UUID, slas []api.SLAUpsert) (*api.SLAResponse, error) {
+	cfg := requestConfig{
+		method: http.MethodPost,
+		url:    fmt.Sprintf("%s/apply-resource-slas/%s", c.routePrefix, resourceID),
+		body:   slas,
+		apiKey: c.apiKey,
```

Comment:
```suggestion
		method: http.MethodPost,
		url:    fmt.Sprintf("%s/apply-resource-slas/%s", c.routePrefix, resourceID),
		body:   slas,
		apiKey: c.apiKey,
		basicAuthKey: c.basicAuthKey,
		successCodes: successCodesStatusOK,
```

I believe we just need the two remaining `requestConfig` fields 👍🏼

Reply:
@mitchnielsen noted on the successCodes, but i intentionally omitted the basicAuthKey because SLAs are a Cloud feature. would we still need it if that's the case?

https://docs.prefect.io/v3/automate/events/slas

Reply:
got it, i noticed we don't pass it into the requestConfig in some places, so I assumed we were passing it selectively

Reply:
added the successCodes slice here: https://github.com/PrefectHQ/terraform-provider-prefect/pull/411/commits/1e9bf840af334679b2e2b424858cd6477bb8c0bb

Reply:
Then it's most likely blind faith Cursor tabbing 😆